### PR TITLE
DATA-5143. Holes are created when a feed with delay is updated/touched

### PR DESCRIPTION
**Pull request by sandeepSamudrala**

### CODE REVIEW

File: oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java [hunk 1]
```diff
@@ -1306,9 +1306,13 @@ private void updateCoords(String cluster, BundleJob bundle,
                 delay = getDelay((Feed) entity, coord);
             }
 
-            //calculate next start time based on delay.
-            endTime = (delay == null) ? endTime
-                    : EntityUtil.getNextStartTime(coord.getStartTime(), delay, EntityUtil.getTimeZone(entity), endTime);
+            //calculate end time based on delay only for feed with delay.
```
Comment (pavankumar526) [inline oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java]: @sandeepSamudrala  Can you explain how start time of new coord will be calculated

Comment (sandeepSamudrala) [inline oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java]: Start time of the new coord is now + delay. While the old coord end time is set to now, which is creating holes.

Comment (sandeepSamudrala) [inline oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java]: Just for reference. https://github.com/InMobi/falcon/blob/master/oozie/src/main/java/org/apache/falcon/oozie/feed/FeedReplicationCoordinatorBuilder.java#L105
The delay gets added to the start time of the new coordinator.

**Comment by pallavi-rao** [comment]
+1 for the patch.

**Comment by ajayyadava** [comment]
Can you please add a test preferably one which fails without this change and passes with this change?

**Comment by sandeepSamudrala** [comment]
@ajayyadava : Can you review it now?

**Comment by ajayyadava** [comment]
:+1:

**Comment by pavankumar526** [comment]
+1

**Pull request merged**